### ISSUE TITLE
GrokConstructor will not work under tomcat 7 + Java 8

### ISSUE THREAD

**Issue by i5513** [stoerr/GrokConstructor#13]
I had to install java 7, maybe could be said on deployment.txt

Thank you!

**Comment by stoerr** [comment]
What is the ctual problem that occurs? Perhaps it can be fixed?

**Comment by i5513** [comment]
My environment:

Centos 6.x , tomcat and java from base + epel

Here is the log at catalina.out:

```
 GRAVE: Compilation error
 org.eclipse.jdt.internal.compiler.classfmt.ClassFormatException
    at org.eclipse.jdt.internal.compiler.classfmt.ClassFileReader.<init>(ClassFileReader.java:369)
    at org.apache.jasper.compiler.JDTCompiler$1.findType(JDTCompiler.java:223)
    at org.apache.jasper.compiler.JDTCompiler$1.findType(JDTCompiler.java:180)
    at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.askForType(LookupEnvironment.java:97)
    at org.eclipse.jdt.internal.compiler.lookup.UnresolvedReferenceBinding.resolve(UnresolvedReferenceBinding.java:49)
    at org.eclipse.jdt.internal.compiler.lookup.BinaryTypeBinding.resolveType(BinaryTypeBinding.java:126)
    at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.getTypeFromTypeSignature(LookupEnvironment.java:1264)
    at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.getTypeFromVariantTypeSignature(LookupEnvironment.java:1320)
    at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.getTypeArgumentsFromSignature(LookupEnvironment.java:1107)
    at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.getTypeFromTypeSignature(LookupEnvironment.java:1269)
    at org.eclipse.jdt.internal.compiler.lookup.BinaryTypeBinding.createMethod(BinaryTypeBinding.java:476)
    at org.eclipse.jdt.internal.compiler.lookup.BinaryTypeBinding.createMethods(BinaryTypeBinding.java:556)
    at org.eclipse.jdt.internal.compiler.lookup.BinaryTypeBinding.cachePartsFrom(BinaryTypeBinding.java:324)
    at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.createBinaryTypeFrom(LookupEnvironment.java:719)
    at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.createBinaryTypeFrom(LookupEnvironment.java:699)
    at org.eclipse.jdt.internal.compiler.Compiler.accept(Compiler.java:294)
    at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.askForType(LookupEnvironment.java:102)
    at org.eclipse.jdt.internal.compiler.lookup.UnresolvedReferenceBinding.resolve(UnresolvedReferenceBinding.java:49)
    at org.eclipse.jdt.internal.compiler.lookup.BinaryTypeBinding.resolveType(BinaryTypeBinding.java:126)
    at org.eclipse.jdt.internal.compiler.lookup.PackageBinding.getTypeOrPackage(PackageBinding.java:168)
    at org.eclipse.jdt.internal.compiler.lookup.Scope.getPackage(Scope.java:2159)
    at org.eclipse.jdt.internal.compiler.ast.ParameterizedQualifiedTypeReference.internalResolveType(ParameterizedQualifiedTypeReference.java:135)
    at org.eclipse.jdt.internal.compiler.ast.ParameterizedQualifiedTypeReference.resolveType(ParameterizedQualifiedTypeReference.java:339)
    at org.eclipse.jdt.internal.compiler.lookup.SourceTypeBinding.resolveTypeFor(SourceTypeBinding.java:1297)
    at org.eclipse.jdt.internal.compiler.lookup.SourceTypeBinding.fields(SourceTypeBinding.java:616)
    at org.eclipse.jdt.internal.compiler.lookup.SourceTypeBinding.faultInTypesForFieldsAndMethods(SourceTypeBinding.java:594)
    at org.eclipse.jdt.internal.compiler.lookup.CompilationUnitScope.faultInTypes(CompilationUnitScope.java:438)
    at org.eclipse.jdt.internal.compiler.Compiler.process(Compiler.java:736)
    at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:444)
    at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:456)
    at org.apache.jasper.compiler.Compiler.compile(Compiler.java:379)
    at org.apache.jasper.compiler.Compiler.compile(Compiler.java:354)
    at org.apache.jasper.compiler.Compiler.compile(Compiler.java:341)
    at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:662)
    at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:364)
    at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:395)
    at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:339)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
    at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
    at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:748)
    at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:486)
    at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:411)
    at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:338)
    at net.stoerr.grokconstructor.webframework.WebDispatcher.doGet(WebDispatcher.scala:51)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:624)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
    at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
    at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505)
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:170)
    at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
    at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:956)
    at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:423)
    at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1079)
    at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625)
    at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:318)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
    at java.lang.Thread.run(Thread.java:745)

 feb 23, 2016 11:16:45 AM net.stoerr.grokconstructor.webframework.WebDispatcher doGet
 GRAVE: org.apache.jasper.JasperException: No se puede compilar la clase para JSP: 

 Ha tenido lugar un error en la línea: [1] en el fichero java generado: [/usr/share/tomcat/work/Catalina/localhost/GrokConstructor/org/apache/jsp/jsp/frame_jsp.java]
 The type java.util.Map$Entry cannot be resolved. It is indirectly referenced from required .class files

 Stacktrace: for http://xx.xx.xx.xx:8080/GrokConstructor/do/match?null:{

 }   requestid=yfMRPGpU

 Request properties: navigation: <li>
         <a href="/GrokConstructor/do/../">
           About
         </a>
       </li><li>
         <a href="/GrokConstructor/do/construction">
           Incremental Construction
         </a>
       </li><li class="active">
         <strong>
           Matcher
         </strong>
       </li><li>
         <a href="/GrokConstructor/do/automatic">
           Automatic Construction
         </a>
       </li>; requestid: yfMRPGpU; title: Test grok patterns; body: <h4>
       Test grok patterns
     </h4><form class="ym-form ym-full" action="/GrokConstructor/do/match#result" method="post">
       <div class="ym-grid ym-equalize">
       <div class="ym-g75 ym-gl box info">
         <p>This tries to parse a set of given logfile lines with a given
     <a href="http://logstash.net/docs/latest/filters/grok">grok regular expression</a>
     and prints
     the matches for named patterns for each log line. You can also apply a
     <a href="http://logstash.net/docs/latest/filters/multiline">multiline filter</a>
     first.</p><p>Please enter some loglines for which you want to check a grok pattern,
       the grok expression that should match these, mark the pattern libraries you draw your patterns from and then press
     </p><input type="submit" class="save ym-button ym-next" value="Go!" id="submit" name="submit" formtarget="_self"/>
       </div>
       <div class="ym-g25 ym-gr box info">
         <p>
     You can also just try this out with a</p><a href="/GrokConstructor/do/match?randomize" class="ym-button ym-add">
     random example
   </a>
       </div>
     </div><div class="ym-fbox-text">
         <label for="loglines">
           Some log lines you want to match. Choose lines that are as diverse as possible.
         </label><textarea name="loglines" id="loglines" cols="180" rows="20"></textarea>
       </div><div class="ym-fbox-text">
         <label for="pattern">
           The pattern that should match all logfile lines:
         </label> <input type="text" name="pattern" id="pattern" size="180"/>
       </div><div class="ym-fbox-text">
       <label>
         Please mark the libraries of
         <a href="http://logstash.net/docs/latest/filters/grok">grok Patterns</a>
         from
         <a href="http://logstash.net/">logstash</a>
         v.2.1 which you want to use. You probably want to use grok-patterns if you use any of the others, since they rely on the basic patterns defined there.</label>
     </div><div class="ym-fbox-check">
       <input type="checkbox" checked="checked" name="groklibs" id="groklibs-firewalls" value="firewalls"/><label for="groklibs-firewalls">
             <a href="/GrokConstructor/groklib/firewalls">
       firewalls
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-haproxy" value="haproxy"/><label for="groklibs-haproxy">
             <a href="/GrokConstructor/groklib/haproxy">
       haproxy
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-linux-syslog" value="linux-syslog"/><label for="groklibs-linux-syslog">
             <a href="/GrokConstructor/groklib/linux-syslog">
       linux-syslog
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-mcollective-patterns" value="mcollective-patterns"/><label for="groklibs-mcollective-patterns">
             <a href="/GrokConstructor/groklib/mcollective-patterns">
       mcollective-patterns
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-postgresql" value="postgresql"/><label for="groklibs-postgresql">
             <a href="/GrokConstructor/groklib/postgresql">
       postgresql
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-java" value="java"/><label for="groklibs-java">
             <a href="/GrokConstructor/groklib/java">
       java
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-grok-patterns" value="grok-patterns"/><label for="groklibs-grok-patterns">
             <a href="/GrokConstructor/groklib/grok-patterns">
       grok-patterns
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-redis" value="redis"/><label for="groklibs-redis">
             <a href="/GrokConstructor/groklib/redis">
       redis
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-nagios" value="nagios"/><label for="groklibs-nagios">
             <a href="/GrokConstructor/groklib/nagios">
       nagios
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-mongodb" value="mongodb"/><label for="groklibs-mongodb">
             <a href="/GrokConstructor/groklib/mongodb">
       mongodb
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-ruby" value="ruby"/><label for="groklibs-ruby">
             <a href="/GrokConstructor/groklib/ruby">
       ruby
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-mcollective" value="mcollective"/><label for="groklibs-mcollective">
             <a href="/GrokConstructor/groklib/mcollective">
       mcollective
     </a>
           </label><input type="checkbox" checked="checked" name="groklibs" id="groklibs-junos" value="junos"/><label for="groklibs-junos">
             <a href="/GrokConstructor/groklib/junos">
       junos
     </a>
           </label>
     </div><div class="ym-fbox-text">
         <label for="grokadditional">
           You can also provide a library of some additional grok patterns in the same format as the pattern files linked above. On each line you give a pattern name, a space and the pattern. For example: WORD \b\w+\b
         </label><textarea name="grokadditional" id="grokadditional" cols="180" rows="5"></textarea>
       </div><div class="ym-fbox-text">
         <label for="multiline">
           If you want to use logstash's multiline filter please specify the used pattern (can include grok Patterns):
         </label> <input type="text" name="multiline" id="multiline" size="80"/>
       </div><div class="ym-fbox-check">
       <input type="checkbox" name="multilinenegate" id="multilinenegate-negate" value="negate"/><label for="multilinenegate-negate">
             <span>negate the multiline regex</span>
           </label>
     </div>
     </form><a id="result">
       <span/>
     </a><span/>
 org.apache.jasper.JasperException: No se puede compilar la clase para JSP: 

 Ha tenido lugar un error en la línea: [1] en el fichero java generado: [/usr/share/tomcat/work/Catalina/localhost/GrokConstructor/org/apache/jsp/jsp/frame_jsp.java]
 The type java.util.Map$Entry cannot be resolved. It is indirectly referenced from required .class files

 Stacktrace:
    at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:103)
    at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:366)
    at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:466)
    at org.apache.jasper.compiler.Compiler.compile(Compiler.java:379)
    at org.apache.jasper.compiler.Compiler.compile(Compiler.java:354)
    at org.apache.jasper.compiler.Compiler.compile(Compiler.java:341)
    at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:662)
    at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:364)
    at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:395)
    at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:339)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
    at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
    at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:748)
    at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:486)
    at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:411)
    at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:338)
    at net.stoerr.grokconstructor.webframework.WebDispatcher.doGet(WebDispatcher.scala:51)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:624)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
    at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
    at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505)
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:170)
    at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
    at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:956)
    at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:423)
    at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1079)
    at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625)
    at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:318)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
    at java.lang.Thread.run(Thread.java:745)


```

Thank you !

**Comment by stoerr** [comment]
Normally, this should work at Java8 and tomcat7. I'm a bit puzzled by your error message: a ClassFormatException occurs if you compile for a higher Java version and try to execute on a lower Java version. So perhaps you are running your tomcat not on Java 8. Or, what seems even more strange to me - it seems tomcat is using the eclipse internal compiler. How can that be? Perhaps it doesn't understand Java 8, here? Can you run it on Oracle Java 8? Either way, this doesn't seem to be a problem of GrokConstruktor itself.

**Comment by i5513** [comment]
Hello,

Sorry for my delayed reply. Today I have to deploy again GrokConstructor, so I cloned the repo again.

$ javac -version
javac 1.8.0_91
$ java -version
openjdk version "1.8.0_91"
OpenJDK Runtime Environment (build 1.8.0_91-b14)
OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)

I have tried , adding to toolchain to maven:
  <toolchain>
    <type>jdk</type>
    <provides>
      <version>1.8</version>
      <vendor>openjdk</vendor>
    </provides>
    <configuration>
      <jdkHome>/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.91-1.b14.el6.x86_64</jdkHome>
    </configuration>
  </toolchain>

$ mvn clean install

But it now show:

> Error message: java.lang.NullPointerException
> 
> Time: Mon Jun 27 11:42:38 CEST 2016
> 
> RequestId: null
> 
> Request Info:
> OUCH: Trouble logging request: java.lang.NullPointerException
> 
> java.lang.NullPointerException
>   at net.stoerr.grokconstructor.webframework.WebDispatcher.requestid(WebDispatcher.scala:38)
>   at net.stoerr.grokconstructor.webframework.WebDispatcher.doGet(WebDispatcher.scala:48)
>   at javax.servlet.http.HttpServlet.service(HttpServlet.java:624)
>   at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
>   at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
>   at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
>   at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
>   at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
>   at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
>   at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
>   at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
>   at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505)
>   at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:170)
>   at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
>   at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:956)
>   at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
>   at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:423)
>   at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1079)
>   at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625)
>   at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.doRun(AprEndpoint.java:2522)
>   at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.run(AprEndpoint.java:2511)
>   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
>   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
>   at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
>   at java.lang.Thread.run(Thread.java:745)

Downloading oracle jdk 8u92 into /opt/java,and compiling with:
JAVA_HOME=/opt/java/ mvn clean install 

$ /opt/java/bin/javac -version ; /opt/java/bin/java -version
javac 1.8.0_92
java version "1.8.0_92"
Java(TM) SE Runtime Environment (build 1.8.0_92-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.92-b14, mixed mode)

produce the same error:

> Error message: java.lang.NullPointerException
> 
> Time: Mon Jun 27 12:03:30 CEST 2016
> 
> RequestId: null
> 
> Request Info:
> OUCH: Trouble logging request: java.lang.NullPointerException
> 
> java.lang.NullPointerException
>   at net.stoerr.grokconstructor.webframework.WebDispatcher.requestid(WebDispatcher.scala:38)
>   at net.stoerr.grokconstructor.webframework.WebDispatcher.doGet(WebDispatcher.scala:48)
>   at javax.servlet.http.HttpServlet.service(HttpServlet.java:624)
>   at javax.servlet.http.HttpServlet.service(HttpServlet.java:731)
>   at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
>   at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
>   at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
>   at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
>   at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
>   at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
>   at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
>   at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505)
>   at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:170)
>   at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
>   at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:956)
>   at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
>   at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:423)
>   at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1079)
>   at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625)
>   at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.doRun(AprEndpoint.java:2522)
>   at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.run(AprEndpoint.java:2511)
>   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
>   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
>   at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
>   at java.lang.Thread.run(Thread.java:745)

**Comment by stoerr** [comment]
Ah, OK, I see the problem.  I probably fixed it with commit afd177bcd4a3917e0eab0d0abf79fd886632dc41 . Can you please try again?
